From cbdbc8d00d31344fafe00e0fdf984e04e631f7c4 Mon Sep 17 00:00:00 2001
From: TBK <tbk@jjtc.eu>
Date: Wed, 26 Feb 2020 21:12:45 +0100
Subject: [PATCH] Fix Linux build error with musl libc

Commit bf5953c549a6d279977df69ffe89b2ba51460eaf caused a build failure
on non-glibc Linux build environments.  Change the conditionalization
so that __GLIBC_PREREQ will only be used if it is defined.

[ghudson@mit.edu: simplified conditionals; rewrote commit message]

ticket: 8880 (new)
tags: pullup
target_version: 1.18-next
---
 src/util/support/plugins.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/util/support/plugins.c b/src/util/support/plugins.c
index 3329db7dc3..1644d16fd0 100644
--- a/src/util/support/plugins.c
+++ b/src/util/support/plugins.c
@@ -62,8 +62,7 @@
  * dlopen() with RTLD_NODELETE, we weren't going to unload the plugin objects
  * anyway.
  */
-#ifdef __linux__
-#include <features.h>
+#ifdef __GLIBC__PREREQ
 #if ! __GLIBC_PREREQ(2, 25)
 #define dlclose(x)
 #endif
